src/cursor.c

```diff
@@ -769,14 +746,8 @@ cursor_button(struct wl_listener *listener, void *data)
 	/* bindings to the Frame context swallow mouse events if activated */
 	bool triggered_frame_binding = false;
 
-	struct view *view = desktop_node_and_view_at(server,
-		server->seat.cursor->x, server->seat.cursor->y, &node,
-		&sx, &sy, &view_area);
-
-	struct wlr_surface *surface = NULL;
-	if (is_surface(view_area)) {
-		surface = lab_wlr_surface_from_node(node);
-	}
+	struct cursor_context c = get_cursor_context(server);
```

Comment:
Usually I am not much a fan of those single char variable names. I can't think of something better that is not way too long right now though.

Reply:
`ctx` maybe?